src/cpp/src/visual_language/cdpruner/cdpruner.cpp

```diff
@@ -0,0 +1,627 @@
+        std::cout << "[CDPruner]   Overall throughput: "
+                  << (static_cast<double>(total_input_tokens) / duration_us * 1'000'000) << " input tokens/sec"
+                  << std::endl;
+        std::cout << "[CDPruner]   Pruning efficiency: "
+                  << (static_cast<double>(actual_total_tokens) / duration_us * 1'000'000) << " output tokens/sec"
+                  << std::endl;
+    } else {
+        std::cout << "[CDPruner]   Overall throughput: N/A" << std::endl;
+        std::cout << "[CDPruner]   Pruning efficiency: N/A" << std::endl;
+    }
+
+    if (total_input_tokens > 0) {
+        std::cout << "[CDPruner]   Combined pruning ratio: "
+                  << (1.0 - static_cast<double>(actual_total_tokens) / total_input_tokens) * 100 << "%" << std::endl;
+    } else {
+        std::cout << "[CDPruner]   Combined pruning ratio: N/A" << std::endl;
+    }
+
+    std::cout << "[CDPruner] Final result: [" << actual_batch_size << ", " << actual_total_tokens << ", "
+              << actual_hidden_dim << "] from " << frame_count << " frames" << std::endl;
+    std::cout << "+----------------------------------------------------------+" << std::endl;
+}
+
+void CDPruner::validate_config(const Config& config) {
+    if (config.pruning_ratio == 0)
+        return;  // Pruning disabled, no validation needed
+
+    if (config.pruning_ratio < 1 || config.pruning_ratio > 100) {
+        throw std::invalid_argument("pruning ratio must be between 0 and 100");
+    }
+
+    if (config.relevance_weight < 0.0f || config.relevance_weight > 1.0f) {
+        throw std::invalid_argument("relevance weight must be between 0.0 and 1.0");
+    }
+
+    if (config.numerical_threshold < 0.0f) {
+        throw std::invalid_argument("numerical_threshold must be positive");
+    }
+
+    if (config.device.empty()) {
+        throw std::invalid_argument("device cannot be empty");
+    }
+}
+
+bool CDPruner::update_config(const Config& new_config) {
+    try {
+        // Validate the new configuration first
+        validate_config(new_config);
+
+        // Update the configuration
+        m_config = new_config;
+
+        // Reinitialize components with new config
+        m_relevance_calc = RelevanceCalculator(new_config);
+        m_kernel_builder = ConditionalKernelBuilder(new_config);
+        m_dpp_selector = FastGreedyDPP(new_config);
+        return true;
+    } catch (const std::exception& e) {
+        return false;
+    }
+}
+
+void CDPruner::validate_input_tensors(const ov::Tensor& visual_features, const ov::Tensor& text_features) {
+    // Validate visual features
+    if (visual_features.get_shape().size() != 3) {
+        throw std::invalid_argument("Visual features must be 3D tensor [B, N, D]");
+    }
+
+    // Validate text features
+    if (text_features.get_shape().size() != 2) {
+        throw std::invalid_argument("Text features must be 2D tensor [M, D]");
+    }
```

Comment:
Suggest to use OPENVINO_ASSERT to replace throw directly.
And below all `throw` should be replaced

Reply:
Updated.